test/libsolidity/SemanticTest.cpp

```
@@ -271,9 +286,44 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
 		return TestResult::Failure;
 	}
 
+	if (m_gasCostFailure)
+	{
+		AnsiColorized(_stream, _formatted, {BOLD, CYAN})
+			<< _linePrefix << "Gas results missing or wrong, obtained result:" << endl;
+		for (auto const& test: m_tests)
+		{
+			ErrorReporter errorReporter;
+			_stream << test.format(errorReporter, _linePrefix, false, _formatted) << endl;
+			_stream << errorReporter.format(_linePrefix, _formatted);
+		}
+		return TestResult::Failure;
+	}
+
 	return TestResult::Success;
 }
 
+bool SemanticTest::checkGasCostExpectation(TestFunctionCall& io_test, bool _compileViaYul) const
+{
+	// TODO: check other evm versions
+	if (m_evmVersion != EVMVersion::latest())
+		return true;
```

I think you can remove this TODO now. I think that the decision on the channel was to run checks only on the latest version?

Also, maybe it would be better if `soltest`/`isoltest` returned an error when you use `--enforce-gas-test` together with `--evm-version` and do not specify the latest EVM? Then you could make this an assert. Currently the requirements will just be silently ignored.

---

Added assertThrow in validate for `CommonOptions` it throws properly in soltest. Also, in isoltest I added it to be default true in case we are running on latest evm. So, this should be fine now